### PR TITLE
Fix some undefined index notices in the `WordPress.XSS.EscapeOutput` sniff.

### PR DESCRIPTION
**Pull request by jrfnl**
These would be thrown when live coding/running the sniffs on code with parse errors.

Quick fix for https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/issues/248#issuecomment-270635692

/cc @grappler

### CODE REVIEW

File: WordPress/Sniffs/XSS/EscapeOutputSniff.php [hunk 1]
```diff
@@ -208,6 +208,11 @@ public function process( PHP_CodeSniffer_File $phpcsFile, $stackPtr ) {
 
 			if ( T_OPEN_PARENTHESIS === $this->tokens[ $i ]['code'] ) {
 
+				if ( ! isset( $this->tokens[ $i ]['parenthesis_closer'] ) ) {
+					// Live coding or parse error.
+					break;
```
Comment (JDGrimes) [inline WordPress/Sniffs/XSS/EscapeOutputSniff.php]: Usually when PHPCS's core sniffs detect a parse error, they don't just skip out, they add a warning. At least they do in some cases. Should we consider doing the same here?

File: WordPress/Sniffs/XSS/EscapeOutputSniff.php [hunk 2]
```diff
@@ -319,7 +324,12 @@ public function process( PHP_CodeSniffer_File $phpcsFile, $stackPtr ) {
 						$i     = ( $function_opener + 1 );
 						$watch = true;
 					} else {
-						$i = $this->tokens[ $function_opener ]['parenthesis_closer'];
+						if ( isset( $this->tokens[ $function_opener ]['parenthesis_closer'] ) ) {
+							$i = $this->tokens[ $function_opener ]['parenthesis_closer'];
+						} else {
+							// Live coding or parse error.
+							break;
```
Comment (JDGrimes) [inline WordPress/Sniffs/XSS/EscapeOutputSniff.php]: Same as above.

File: WordPress/Sniffs/XSS/EscapeOutputSniff.php (hunk 1 above)
Comment (jrfnl) [inline WordPress/Sniffs/XSS/EscapeOutputSniff.php]: IMHO that's what the `Generic.PHP.Syntax` sniff is for (included in `extra`)

Comment (JDGrimes) [inline WordPress/Sniffs/XSS/EscapeOutputSniff.php]: Yep, that makes sense. So I guess for consistency we should change [the one place](https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/blob/51eb33b15f3aad5d74e1c816b5d8edc42163de25/WordPress/Sniffs/PHP/StrictInArraySniff.php#L103) we give an error for syntax issues. But that can be a separate PR, I guess.
